packages/@aws-cdk/aws-apigateway/lib/model.ts

```diff
@@ -1,46 +1,151 @@
-export interface IModel {
-    readonly modelId: string;
+import { Construct, IResource, Resource } from '@aws-cdk/core';
+import { CfnModel, CfnModelProps } from './apigateway.generated';
+import jsonSchema = require('./json-schema');
+import { IRestApi, RestApi } from './restapi';
+
+export interface IModel extends IResource {
+  /**
+   * A name for the model. If you don't specify a name, AWS CloudFormation
+   * generates a unique physical ID and uses that ID for the model name.
+   * For more information, @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-name.html.
+   *
+   * Important
+   * If you specify a name, you cannot perform updates that require replacement
+   * of this resource. You can perform updates that require no or some interruption.
+   * If you must replace the resource, specify a new name.
+   *
+   * @attribute
+   */
+  readonly modelName: string;
```

Comment:
Hey, we aren't able to introduce breaking changes anymore. Please revert to `modelId` please.

Reply:
Done (although technically the ref gives the modelName not the modelId in CloudFormation terms)

Reply:
Semantically it's the same thing. No one expects the modelId to be the actual Logical Id, and the only references to the model ate through the name (in requests and responses)